docs: add recommendation to use closures for configurable extensions (#7048)
vuestorefront/vue-storefront: docs/content/3.middleware/2.guides/3.extensions.md

> Extensions can be added to any integration in your `middleware.config.ts` file by creating an `extensions` function. This function accepts an array of extensions and returns a new array with your extension added.

Reviewer: ```suggestion
Extensions can be added to any integration in your `middleware.config.ts` file by creating an `extensions` function. This function receives an array of existing extensions as the only argument and has to return an updated array of extensions (with your extension added).
```

> You can use this function to add your own extensions or to modify existing ones.

Reviewer: This can be deleted in case you accept the suggestion above

```js
export const integrations = {
  magento: {
    // ...
    extensions: (extensions) => [

>       ...extensions, // don't forget to add existing extensions

Reviewer: ```suggestion
      ...extensions, // don't forget to include existing extensions
```

  -H 'content-type: application/json' \
  -d '[{ "id": 1, "name": "John" }]'
```

### Creating a configurable extension

In more complex cases, you might want to create an extension that can be configured.

Imagine that you're preparing an extension that would add cache control headers to responses based on the configuration.
It might look like this:

> ```ts

Reviewer 2: I would recommend either removing the code snippet of the "non-recommended" way or having a comment at the top that says that this is the less reusable way/suboptimal to do this.

That way someone doesn't start reading this code snippet thinking it's what we recommend and copy/pasting it into their project before finishing the doc

Author: @mattmaribojoc suggestion applied!

  return {
    name: "caching-extension",
    hooks(req, res) {
      return {
        afterCall({ response, callName }) {
          if (req.method !== "GET") {
            console.log("[CACHING] It's not a GET request, skipping caching");
            return response;
          }

          if (res.getHeader("set-cookie")) {
            console.log(
              "[CACHING] Response containing Set-Cookie header, skipping caching"
            );
            return response;
          }

          if (callName in config) {
            res.set("Cache-Control", config[callName]);
          }

          return response;
        },
      };
    },
  };
};
```

Then, you can use it in your `middleware.config.ts` file:

```ts
export const integrations = {
  sapcc: {
    // ...
    extensions: (extensions) => [
      ...extensions,
      createCacheControlExtension({
        searchProducts: "public, max-age=3600",
      }),
    ],
  },
  contentful: {
    // ...
    extensions: (extensions) => [
      ...extensions,
      createCacheControlExtension({
        getEntries: "public, max-age=3600",
      }),
    ],
  },
};
```

> Using closures is a recommended way to create a configurable extension. It allows you to reuse the same logic in multiple integrations and keeps your code clean and maintainable.

Reviewer 2: I think we can move this paragraph before the closure code snippet.

Author: @mattmaribojoc suggestion applied!

  -H 'content-type: application/json' \
  -d '[{ "id": 1, "name": "John" }]'
```

### Creating a configurable extension

In more complex cases, you might want to create an extension that can be configured.

To avoid code duplication, you can prepare a closure that will create a new extension with a given configuration.

Using closures is a recommended way to create a configurable extension. It allows you to reuse the same logic in multiple integrations and keeps your code clean and maintainable.

```ts
const createCacheControlExtension = (config: Record<string, string>) => {
  return {

>     name: "caching-extension",

Reviewer: nitpick
```suggestion
    name: "cache-control-extension",
```
